File: Arcmage.Web/src/app/models/card-search-options.ts
import { SearchOptionsBase } from "./search-options-base";
import { CardType } from "./card-type";
import { Faction } from "./faction";
import { Serie } from "./serie";
import { RuleSet } from "./rule-set";
import { Status } from "./status";
import { Language } from "./language";

export class CardSearchOptions extends SearchOptionsBase {
   cost: string;
   cardType: CardType;
   faction: Faction;
   serie: Serie;
   ruleSet: RuleSet;
   status: Status;
   loyalty: number | null;
   language: Language;
}

File: Arcmage.Model/CardSearchOptions.cs
namespace Arcmage.Model
{
    public class CardSearchOptions : SearchOptionsBase
    {

        public string Cost { get; set; }

        public CardType CardType { get; set; }

        public Faction Faction { get; set; }

        public Serie Serie { get; set; }

        public RuleSet RuleSet { get; set; }

        public Status Status { get; set; }

        public Language Language { get; set; }

        public int? Loyalty { get; set; }

    }
}
